File: daily_problems/2026/04/0425/personal_submission/cf106495h_yawn_sean.py
# Submission link: https://codeforces.com/gym/106495/submission/372310512
def main():
    t = II()
    outs = []
    
    for _ in range(t):
        s1 = [ord(c) % 26 for c in I()]
        s2 = [ord(c) % 26 for c in I()]
        
        n = len(s1)
        s = [0] * n
        
        for i in range(n):
            s[i] = s1[i] * 26 + s2[i]
        
        tmp = [s[i // 2] if i % 2 == 0 else -1 for i in range(2 * n - 1)]
    
        manacher = [0] * (2 * n - 1)
        starting = [0] * (2 * n)
        ending = [0] * (2 * n)
        
        j = 0
        for i in range(2 * n - 1):
            if j + manacher[j] >= i:
                manacher[i] = fmin(manacher[2 * j - i], j + manacher[j] - i)
            while i - manacher[i] - 1 >= 0 and i + manacher[i] + 1 < 2 * n - 1 and tmp[i - manacher[i] - 1] == tmp[i + manacher[i] + 1]:
                manacher[i] += 1
            if i + manacher[i] > j + manacher[j]:
                j = i
            
            starting[i - manacher[i]] += 1
            starting[i + 1] -= 1
            
            ending[i] += 1
            ending[i + manacher[i] + 1] -= 1
    
        for i in range(1, 2 * n):
            starting[i] += starting[i - 1]
            ending[i] += ending[i - 1]
        
        ans = 0
        for i in range(n):
            if s1[i] != s2[n - 1 - i] or s2[i] != s1[n - 1 - i]:
                break
            ans += 1
            if i < n - 1:
                ans += starting[2 * (i + 1)]
                ans += ending[2 * (n - i - 2)]
        
        outs.append(ans)
    
    print('\n'.join(map(str, outs)))

File: daily_problems/2026/04/0425/personal_submission/cf106495h_yawn_sean.cpp
#include <bits/stdc++.h>
// #pragma GCC optimize("O3,Ofast,unroll-loops")
// #include "atcoder/all"

using namespace std;
auto rng = mt19937(random_device()());
auto rngl = mt19937_64(random_device()());

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(0);
	cout.tie(0);

	int t;
	cin >> t;

	while (t --) {
		string s1, s2;
		cin >> s1 >> s2;

		int n = s1.size();
		vector<int> tmp;

		for (int i = 0; i < n; i ++) {
			if (i) tmp.emplace_back(-1);
			tmp.emplace_back(s1[i] * 1000 + s2[i]);
		}

		vector<int> manacher(2 * n - 1, 0), starting(2 * n, 0), ending(2 * n, 0);
		int j = 0;

		for (int i = 0; i < 2 * n - 1; i ++) {
			if (j + manacher[j] >= i)
				manacher[i] = min(manacher[2 * j - i], j + manacher[j] - i);
			while (i - manacher[i] - 1 >= 0 && i + manacher[i] + 1 < 2 * n - 1 && tmp[i - manacher[i] - 1] == tmp[i + manacher[i] + 1])
				manacher[i] ++;
			if (i + manacher[i] > j + manacher[j]) j = i;

			starting[i - manacher[i]] ++;
			starting[i + 1] --;

			ending[i] ++;
			ending[i + manacher[i] + 1] --;
		}

		for (int i = 1; i < 2 * n; i ++) {
			starting[i] += starting[i - 1];
			ending[i] += ending[i - 1];
		}

		long long ans = 0;

		for (int i = 0; i < n; i ++) {
			if (s1[i] != s2[n - 1 - i] || s2[i] != s1[n - 1 - i]) break;
			ans ++;
			if (i < n - 1) {
				ans += starting[2 * (i + 1)];
				ans += ending[2 * (n - i - 2)];
			}
		}

		cout << ans << '\n';
	}

	return 0;
}